bot/cogs/starboard.py

```diff
@@ -0,0 +1,540 @@
+import asyncio
+import logging
+from textwrap import dedent
+from typing import Optional, Tuple
+
+from aiohttp import ClientResponseError
+from discord import (
+    Colour, Embed, Forbidden, HTTPException,
+    Message, RawReactionActionEvent, TextChannel
+)
+from discord.ext.commands import Bot, CommandError, command
+from discord.utils import get
+
+from bot.constants import Channels, Colours, Guild, Keys, Roles, URLs
+from bot.decorators import with_role
+
+
+LVL1_STAR = "\u2b50"
+LVL2_STAR = "\U0001f31f"
+LVL3_STAR = "\U0001f4ab"
+LVL4_STAR = "\u2728"
+
+YES_EMOJI = "\u2705"
+NO_EMOJI = "\u274e"
+OK_HAND = "\U0001f44c"
+
+THRESHOLDS = {
+    LVL1_STAR: 2,
+    LVL2_STAR: 5,
+    LVL3_STAR: 10,
+    LVL4_STAR: 20
+}
+
+ALLOWED_TO_STAR = (Roles.admin, Roles.moderator, Roles.owner, Roles.helpers)
```

Comment:
My opinion is that this is not suited for mass clicks as it doesn't have any form of rate limiting the checks.  If someone spams a reaction it will spam a message edit, at least 2 api requests and other stuff.  For it to be public some form of rate limiting needs to be implemented imo